lib/page/profile.dart

```diff
@@ -221,9 +245,9 @@ class _ProfileState extends State<Profile> {
     );
   }
 
-  List<Widget> buildInterest() {
+  List<Widget> buildInterest(List<PreferenceModel> preferences) {
     List<Widget> widgets = [];
-    for (String interest in interests) {
+    for (PreferenceModel preference in preferences) {
```

Comment:
use map() instead of adding them to widgets

Reply:
done, now it uses map